tests/model_inheritance/tests.py

```diff
@@ -149,6 +150,25 @@ def test_custompk_m2m(self):
         # accidentally found).
         self.assertSequenceEqual(s.titles.all(), [])
 
+    def test_create_diamond_mti_common_parents(self):
+        with self.assertNumQueries(5):
+            itrmp = ItalianRestaurantManyParents.objects.create(
+                name="Ristorante Miron",
+                address="1234 W. Ash",
+            )
+
+        italian_resturant_place = itrmp.italianrestaurant_ptr.place_ptr
+        itrmp_place = itrmp.place_ptr_2
+        italian_resturant_resturant = itrmp.italianrestaurant_ptr.restaurant_ptr
+        itrmp_resturant = itrmp.restaurant_ptr
+
+        # Some draft tests to test regression
+        self.assertEqual(italian_resturant_place, itrmp_place)
+        self.assertEqual(italian_resturant_resturant, itrmp_resturant)
+        self.assertEqual(itrmp_resturant.place_ptr, itrmp_place)
+        self.assertEqual(itrmp.name, "Ristorante Miron")
```

Comment:
This failure is completely unexpected, cannot reproduce or figure out the reason in local.